### PR TITLE
EZP-28863: REST search error for searching fields with the same name

### PR DESCRIPTION
**Pull request by ViniTou**
| Question           | Answer
| ------------------ | ------------------
| **JIRA issue**     | [EZP-28863](https://jira.ez.no/browse/EZP-28863)
| **Bug/Improvement**| yes
| **New feature**    | no
| **Target version** | `master`
| **BC breaks**      | no
| **Tests pass**     | yes
| **Doc needed**     | no

Not sure if should target master or 7.x, it is bugfix but for relatively new feature.

**TODO**:
- [x] Implement feature / fix a bug.
- [x] Implement tests.
- [x] Fix new code according to Coding Standards (`$ composer fix-cs`).
- [x] Ask for Code Review.

### CODE REVIEW

**Comment by andrerom** [comment]
@ViniTou you can target 7.1 here now ;)

**Comment by ViniTou** [comment]
@andrerom 
@alongosz 

All green now :)

File: eZ/Publish/API/Repository/Tests/SearchServiceTest.php [hunk 1]
```diff
@@ -4499,4 +4499,95 @@ private function getContentInfoFixtureClosure($closure = null)
             }
         };
     }
+
+    /**
+     * Test for the findContent() method.
+     *
+     * @see \eZ\Publish\API\Repository\SearchService::findContent()
+     */
+    public function testNonImplementedLegacySearchNonBlock()
```
Comment (alongosz) [inline eZ/Publish/API/Repository/Tests/SearchServiceTest.php]: I honestly have no idea what that means :)
maybe `testFieldCriterionForContentsWithIdenticalFieldIdentifiers`
and elaborate a bit more in PhpDoc:
```
Test searching using Field Criterion where the given Field Identifier exists in both searchable and non-searchable Fields.
```

File: eZ/Publish/API/Repository/Tests/SearchServiceTest.php [hunk 2]
```diff
@@ -4499,4 +4499,95 @@ private function getContentInfoFixtureClosure($closure = null)
...
+        $this->assertEquals(
+            $testContent->id,
+            $result->searchHits[0]->valueObject->id
+        );
+    }
+
+    private function createContentWithFieldType(
+        string $fieldType,
+        string $fieldName,
+        string $fieldContent
```
Comment (alongosz) [inline eZ/Publish/API/Repository/Tests/SearchServiceTest.php]: Naming: `$fieldValue`.

File: eZ/Publish/Core/Search/Legacy/Content/Common/Gateway/CriterionHandler/Field.php [hunk 3]
```diff
@@ -154,11 +159,21 @@ public function handle(
         );
 
         $whereExpressions = array();
+
+        $columnInfoArray = array_column($fieldsInformation, 'column');
+
+        if (false === reset($columnInfoArray) && 1 === count(array_unique($columnInfoArray))) {
+            throw new NotImplementedException(
```
Comment (alongosz) [inline eZ/Publish/Core/Search/Legacy/Content/Common/Gateway/CriterionHandler/Field.php]: This whole block looks rather complicated. Not that much, but wouldn't we achieve the same result checking if `$whereExpressions` is empty after the next loop?

**Pull request merged**